components/log-viewer-webui/client/webpack.config.js

```diff
@@ -0,0 +1,82 @@
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import MiniCssExtractPlugin from "mini-css-extract-plugin";
+import * as path from "node:path";
+import {fileURLToPath} from "node:url";
+
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+const isProduction = "production" === process.env.NODE_ENV;
+
+const stylesHandler = isProduction ?
+    MiniCssExtractPlugin.loader :
+    "style-loader";
+
+const plugins = [
+    new HtmlWebpackPlugin({
+        template: path.resolve(dirname, "public", "index.html"),
+    }),
+];
+
+const config = {
+    devtool: isProduction ?
+        "source-map" :
+        "eval-source-map",
+    entry: path.resolve(dirname, "src", "index.jsx"),
+    module: {
+        rules: [
+            {
+                test: /\.(js|jsx)$/i,
+                exclude: /node_modules/,
+                use: {
+                    loader: "babel-loader",
+                    options: {
+                        presets: [
+                            "@babel/preset-env",
+                            [
+                                "@babel/preset-react",
+                                {
+                                    runtime: "automatic",
+                                },
+                            ],
+                        ],
+                    },
+                },
+            },
+            {
+                test: /\.css$/i,
+                use: [
+                    stylesHandler,
+                    "css-loader",
+                ],
+            },
+        ],
+    },
+    output: {
+        path: path.resolve(dirname, "dist"),
+        filename: "[name].[contenthash].bundle.js",
```

Comment:
The contenthash has been causing issues with HMR / Fast Refresh in my experience and I recommend removing such in a dev environment setting.
```suggestion
        filename: isProduction?
            "[name].[contenthash].bundle.js":
            "[name].bundle.js",
```